pkg/tarmak/cmd.go

```diff
@@ -82,10 +84,12 @@ func (c *CmdTarmak) Apply() error {
 	default:
 	}
 
-	// wait for convergance in every mode
-	err := c.Cluster().WaitForConvergance()
-	if err != nil {
-		return err
+	// wait for convergance if flag enabled and has changed
+	if hasChanged && c.flags.Cluster.Apply.WaitForConvergence {
```

Comment:
We need to wait for convergance when we update the config through the tarmak update method  (and not terraform) in ConfigurationOnly 

We update the puppet tar config and send out a converge request. The only way to skip that would if we would know if the config has changed through our upload method in tarmak (puppet.tar.gz is the same).

When we update the puppet.tar.gz in the default mode we have a terraform change for updating the puppet.tar.gz